go.mod

```diff
@@ -0,0 +1,10 @@
+module github.com/GetStream/stream-go2/v4
```

Comment:
why would we include /v4 to the module name? sounds redundant

Reply:
There are already 3.x releases and it's best practice to bump major while introducing modules.

Reply:
yes but why are we adding the version to the name of the module? can't we just create a v4.0.0 tag?

Reply:
no

Reply:
`module github.com/GetStream/stream-go2`

Reply:
How do you differentiate non-module users with module ones?

How do you bump to v5 later?

If it's v4 (above v2+), it should be added to module path.

Reply:
I am sure we can find a solution that does not require having version name as part of module and package names; that's ugly

Reply:
It's the convention of go tool works with major versions.

I think it's not ugly, but explicit.

If you really go tag way only, I can update similar to stripe-go but beware not standard in terms of introducing breaking changes. 

https://github.com/stripe/stripe-go#go-module-support

Reply:
I came across with wrong module support addition without version suffix in the module name.

https://github.com/mholt/archiver/issues/187

should be `github.com/mholt/archiver/v4` but it's not so it's breaking and module supported go can only install `v3.1.1` - last version before `go.mod` file was added.

Reply:
Just passing by (got here via the linked module issue in mholt/archiver), but wanted to note that @ferhatelmas is correct -- Go modules with a major version beyond v1 are required to have the major version as a suffix of their module name per the module specification. See https://github.com/golang/go/wiki/Modules#releasing-modules-v2-or-higher, which discusses this exact scenario (transitioning a repository that is already at v2+ to modules). https://blog.golang.org/v2-go-modules discusses this as well. As noted here, not including the major version as part of the module while having tags that are beyond v0/v1 will result in suboptimal behavior when trying to consume the module (as is currently the case with archiver). Unfortunately this issue/confusion appears to be quite common: https://github.com/golang/go/issues/31543

Reply:
@nmiyake thanks for chiming in to give further clarification and links. 

Also thanks for the archiver fix :wink: